src/plugins/data/server/index_patterns/routes/util/schemas.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const serializedFieldFormatSchema = schema.object({
+  id: schema.maybe(schema.string()),
+  params: schema.maybe(schema.any()),
+});
+
+export const fieldSpecSchemaFields = {
+  name: schema.string({
+    maxLength: 1_000,
+  }),
+  type: schema.string({
+    defaultValue: 'string',
+    maxLength: 1_000,
+  }),
+  searchable: schema.boolean({ defaultValue: false }),
+  aggregatable: schema.boolean({ defaultValue: false }),
+  count: schema.maybe(
+    schema.number({
+      min: 0,
+    })
+  ),
+  script: schema.maybe(
+    schema.string({
+      maxLength: 1_000_000,
+    })
+  ),
+  lang: schema.maybe(
```

Comment:
I think the only valid value for this is `painless`, therefore it doesn't need to be set.